### PR TITLE
Nt pixel scale

### PR DESCRIPTION
**Pull request by ntellis**

### CODE REVIEW

**Comment by ntellis** [comment]
This PR changes the plotting functionality in Cutouts to read the pixel scale from the WCS (World coordinate system) values in the fits header of retrieved images, and use arcsecond rather than pixel values as the user input.

Note that our expectation is that a particular SIA service will provide fits images with the correct header info.

Also notably, NSC's SIA cutout service differs from SkyMapper's SIA cutout service in the format of the result table returned. This implies that a dataset-specific handler to coerce a particular SIA response to a common format will be required to have the core plotting utilities be general.

File: cutouts/io.py [hunk 1]
```diff
@@ -88,7 +89,13 @@ def find_cutout(
         using this particular SIA Service.
     """
     center = (ra, dec)
-    result = sia_service.search(center, size=(height/3600., width/3600.)).to_table().to_pandas()
+
+    result = sia_service.search(center, size=(height/3600., width/3600.))
+    if isinstance(result, SIAResults):
```
Comment (akoumjian) [inline cutouts/io.py:94]: What's going on here? Is it the `SIAService` class itself that returns different types depending on which server you query? If so, we can wrap that class to provide a consistent interface and put this conditional inside it?

File: cutouts/plot.py [hunk 2]
```diff
@@ -367,9 +375,11 @@ def plot_cutout(
     image = hdu.data
     hdr = hdu.header
     wcs = WCS(hdr)
-
-    image_centered, x_offset, y_offset = center_image(image, wcs, ra, dec, height=height, width=width)
-
+    pixel_scales = proj_plane_pixel_scales(wcs)
```
Comment (akoumjian) [inline cutouts/plot.py:378]: nitpick, but can we unpack these into something like `width_pixel_scale, height_pixel_scale` rather than using indexing?

**Review by akoumjian — CHANGES_REQUESTED** [review]

**Comment by ntellis** [comment]
Recent commits include the changes suggested by @akoumjian 

Additionally, crosshairs / arrow dimensions are now specified in arcseconds and scaled according to WCS. This seemed to make sense, given that now the entry point to Cutouts assumes no knowledge of the number of pixels returned per arcsecond for a given cutout provider. It also lets us use the crosshairs as a visual scale, as below (skymapper cutout of Edlu from 2014, 30"x30", and a sample from nsc, 20"x20"). Note the velocity vector in the NSC case is bogus.

skymapper_edlu_30
Screenshot 2023-02-17 at 5 12 31 PM

The other thought I had was to just scale the crosshairs/arrow proportionally to the image size, and happy to change over to that if that seems more intuitive.

**Comment by moeyensj** [comment]
This is a great change!

**Comment by KatKiker** [comment]
I have some reservations about changing the size of the arrow with the scale. Usually when one velocity vector is larger than another, it means the magnitude is greater. In the above images, for example, most people would read this as the top asteroid moving slower than the bottom, not that they had different arcsecond scales. 

I don't think it would matter as long as all the cutouts are the same arcsecond length though.

**Comment by ntellis** [comment]
> I have some reservations about changing the size of the arrow with the scale. Usually when one velocity vector is larger than another, it means the magnitude is greater. In the above images, for example, most people would read this as the top asteroid moving slower than the bottom, not that they had different arcsecond scales.
> 
> I don't think it would matter as long as all the cutouts are the same arcsecond length though.

I'll adjust this so that the velocity vector is scaled to the image size instead of being specified in arcseconds. Your point is well taken that it seems to imply magnitude.

File: cutouts/sia.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        else:
+            result = result.to_table().to_pandas()
+        # Filter out non-image results
+        if "prodtype" in result.columns:
+            result = result[result["prodtype"] == "image"]
+
+        # Normalize access_url column
+        if "access_url" not in result.columns:
+            if "get_fits" in result.columns:
+                result["access_url"] = result["get_fits"]
```
Comment (ntellis) [inline cutouts/sia.py:71]: @akoumjian Specifically these conditionals were added to support the different fields we get back from the two SIA services.

**Comment by moeyensj** [comment]
This was merged in #15. @ntellis are you okay if I close this?

**Comment by ntellis** [comment]
A-OK

**Pull request closed without merging**